compiler/tests/generated/generic_service_grpc.h

```diff
@@ -35,7 +35,8 @@
 namespace tests
 {
 
-class Foo final
+template <typename Payload>
+    class Foo final
 {
 public:
     template <typename TThreadPool>
```

Comment:
Well, this service in particular is a generic service (https://github.com/lalo/bond/blob/7329aaa47da678289122d13cfa1e3811ea048515/examples/cpp/grpc/pingpong/pingpong.bond#L26) so that is why we have a template here. 

I stuck with std::function from the beginning, I think we could do an arbitrary object since our thread_pool supports it (https://github.com/Microsoft/bond/blob/grpc%2B%2B/cpp/inc/bond/ext/grpc/thread_pool.h#L91)